core/msg_calculate.go

```diff
@@ -441,8 +443,16 @@ func DoCalculate(quit <-chan struct{}, ctx *Context, req *CalculateRequest, c ip
 	}
 	wait.Wait()
 
-	if quit != ctx.Rollback.GetChannel() {
-		return &CalcCancelByRollbackError{blockHeight}, blockHeight, nil, nil
+	if quit != ctx.CancelCalculation.GetChannel() {
+		var err error
+		switch ctx.CancelCalculation.cancelCode {
+		case CancelExit:
+			return &CalcCancelByExit{blockHeight}, blockHeight, nil, nil
+		case CancelRollback:
+			err = &CalcCancelByRollbackError{blockHeight}
```

Comment:
Why handle two case differently?

ctx.CancelCalculation.cancelCode is not referenced in other code